src/main/scala/scorex/core/network/NetworkController.scala

```diff
@@ -36,8 +36,8 @@ class NetworkController(settings: NetworkSettings,
 
   import NetworkController._
 
-  private val synchronizerProps = Props(new PeerSynchronizer(self, peerManagerRef, settings))
-  private val peerSynchronizer = context.system.actorOf(synchronizerProps, "PeerSynchronizer")
+  private val synchronizerProps: Props = PeerSynchronizerRef.props(self, peerManagerRef, settings)
```

Comment:
You can replace lines 39 and 40 by:

```
private val peerSynchronizer = PeerSynchronizerRef(self, peerManagerRef, settings)
```

The apply method in `PeerSynchronizerRef` already creates the props and inserts it into the implicit `context.system`.

We don't need explicit types (e.g. `ActorRef`) when the `val` is private and the type is obvious.

Reply:
Good catch, I didn't notice the line

```
  private implicit val system: ActorSystem = context.system
```
below those. I will update this.